test/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Tests/ExtensionsTests/StringExtensionTest.cs

```diff
@@ -45,10 +45,61 @@ public static IEnumerable<object[]> GetTestCasesForStringExtensionShouldSucceed(
             };
         }
 
+        public static IEnumerable<object[]> GetTestCasesForRemoveRoslynDuplicateStringShouldSucceed()
+        {
+            yield return new object[]
+            {
+                "String with special character $",
+                "$skip$skip",
+                "$skip"
+            };
+
+            yield return new object[]
+            {
+                "String with special character @",
+                "@service_Catalog_Id@Id@test@Id@test",
+                "@service_Catalog_Id@Id@test"
+            };
+
+            yield return new object[]
+            {
+                "String with special character @ and duplicate not introduced by roslyn",
+                "@skip@skip",
+                "@skip@skip"
+            };
+
+            yield return new object[]
+            {
+                "String with special character @ and no duplicate",
+                "@skip",
+                "@skip"
+            };
+
+            yield return new object[]
+            {
+                "String with multiple special characters",
+                "service-$Catalog-Id-1-$Catalog-Id-1",
+                "service-$Catalog-Id-1"
+            };
+
+            yield return new object[]
+            {
+                "String with special character -",
+                "ms-cv-cv",
+                "ms-cv"
+            };
+
+            yield return new object[]
+            {
+                "string with no special character",
+                "testParam",
+                "testParam"
+            };
+        }
 
         [Theory]
-        [MemberData(nameof(GetTestCasesForStringExtensionShouldSucceed))]
-        public void StringExtensionShouldUpdateCorrectly(
+        [MemberData(nameof(GetTestCasesForToCamelCaseShouldSucceed))]
```

Comment:
Let's use "UpdateCorrectly" instead of "Succeed" for consistency. Applies to the method below as well.